pkgs/development/python-modules/shapely/default.nix

```diff
@@ -45,7 +45,10 @@ buildPythonPackage rec {
   ];
 
   preCheck = ''
-    rm -r shapely # prevent import of local shapely
```

Comment:
This is ugly but without it, test collection fails because it will try to load shapely from the project directory:

```
============================= test session starts ==============================
platform linux -- Python 3.11.5, pytest-7.4.2, pluggy-1.2.0
rootdir: /build/shapely-2.0.2
configfile: setup.cfg
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/config/__init__.py:641: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/pathlib.py:567: in import_path
    importlib.import_module(module_name)
/nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1126: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1126: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1126: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:940: in exec_module
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
shapely/__init__.py:1: in <module>
    from shapely.lib import GEOSException  # NOQA
E   ModuleNotFoundError: No module named 'shapely.lib'
=========================== short test summary info ============================
ERROR  - ModuleNotFoundError: No module named 'shapely.lib'
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.08s ===============================
```

Reply:
It is related to this issue, https://github.com/NixOS/nixpkgs/issues/255262.
Changing the current directory is one of the workarounds.

Reply:
I noticed that `cd $out` works but decided against it because then it would run tests installed to `$out` and I am not sure all tests are installed (and if they should be).

Reply:
As the main investigator of #255262, I personally prefer `cd $out` - if we suspect upstream doesn't install all tests, perhaps we can check that by listing what gets installed in the package vs what's in the source? I doubt upstream won't install either all of none.

In any case, a link to https://github.com/NixOS/nixpkgs/issues/255262 near the workaround (which ever you'd eventually choose) would be great.